test/tests/kernels/2d_diffusion/matdiffusion.i

```diff
@@ -0,0 +1,75 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 2
+  ny = 2
+  xmax = 1.0
+  ymax = 1.0
+  elem_type = QUAD4
+[]
+
+[Variables]
+  [./u]
+  [../]
+[]
+
+[Kernels]
+  [./cres]
+    type = MatDiffusion
+    variable = u
+    D_name = Du
+  [../]
+  [./ctime]
+    type = TimeDerivative
+    variable = u
+  [../]
+[]
+
+[Materials]
+  [./Dc]
+    type = DerivativeParsedMaterial
+    f_name = Du
+    function = '0.01+u^2'
+    args = 'u'
+    derivative_order = 1
+  [../]
+[]
+
+[BCs]
+  [./left]
+    type = DirichletBC
+    variable = u
+    boundary = 1
+    value = 0
+  [../]
+
+  [./right]
+    type = NeumannBC
+    variable = u
+    boundary = 2
+    value = 1
+  [../]
+[]
+
+[Preconditioning]
+  [./SMP]
+    type = SMP
+    full = true
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  scheme = 'BDF2'
+
+  solve_type = 'NEWTON'
+  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_pc_type -pc_asm_overlap'
+  petsc_options_value = 'asm      31                  lu           1'
```

Comment:
The problem seems to solve equally well without these options.

Reply:
Sure, I removed those options.